agent/agent_endpoint.go

```diff
@@ -244,7 +244,7 @@ func (s *HTTPHandlers) AgentReload(resp http.ResponseWriter, req *http.Request)
 		return nil, acl.ErrPermissionDenied
 	}
 
-	return nil, s.agent.ReloadConfig()
```

Comment:
Trying to wrap my head around this -- why do we say `false` here ?

Reply:
The main reason for the flag is to say if we are calling the `ReloadConfig` manually or automatically through auto-config-reload. I can maybe create a wrapper that would underneath call the function with the right flag value to make it more readable.

Reply:
`AutoReloadConfig` and `ReloadConfig` could both call `reloadConfig(bool)`